crates/turborepo-repository/src/change_mapper.rs

```diff
@@ -8,56 +8,120 @@ use wax::Program;
 
 use crate::package_graph::{ChangedPackagesError, PackageGraph, PackageName, WorkspacePackage};
 
+pub enum PackageDetection {
+    /// We've hit a global file, so all packages have changed
+    All,
+    /// This change is meaningless, no packages have changed
+    None,
+    /// This change has affected one package
+    Package(WorkspacePackage),
+}
+
+/// Detects which packages are affected by a file change. This can be a single
+/// package (`Package`), none of the packages (`None`), or all of the packages
+/// (`All`).
+pub trait PackageDetector {
+    fn detect_package(&self, file: &AnchoredSystemPath) -> PackageDetection;
+}
```

Comment:
nit: I don't love the naming here. In english, this trait/method isn't detecting packages (which sounds more like what `turbo info` does), it is inferring how a file change affects a known list of packages. Just a suggestion:

```suggestion
pub trait WorkspaceChangeDetector {
    fn get_changed_package(&self, file: &AnchoredSystemPath) -> WorkspaceChange;
}
```

Reply:
Hmm yeah agreed. I do feel like WorkspaceChangeDetector is a little confusing too. Maybe PackageChangeMapper? It's mapping a single change to a package

Reply:
yeah I didn't want to use "change mapper" since we use that upstream already 😓 . In lieu of an obvious answer, no need to make changes :)